lib/SyTest/Federation/Server.pm

```diff
@@ -225,6 +226,67 @@ sub on_request_key_v2_server
    } ) );
 }
 
+sub on_request_federation_v1_query_directory
+{
+   my $self = shift;
+   my ( $req, $alias ) = @_;
+
+   my $room_id = $self->{datastore}->get_alias( $alias ) or
+      return Future->done( response => HTTP::Response->new(
+         404, "Not found", [ Content_length => 0 ], "",
+      ) );
+
+   Future->done( json => {
```

Comment:
To tell the responder to JSON encode this result. Not all results are JSON-encoded (because of the media server, etc...)

https://github.com/matrix-org/sytest/blob/develop/lib/SyTest/Federation/Server.pm#L109